src/SimpleNameRegister.sol

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+contract SimpleNameRegister {
+    
+    // map a string to an address to identify current owner
+    mapping (string => address) public nameOwner;    
+
+    // emit event when name is registered or relinquished
+    event NameRegistered(address indexed owner, string indexed name);
+    event NameRelinquished(address indexed owner, string indexed name);
```

Comment:
I'm not sure that you can index strings in events, would you check?

Reply:
Fundamentally, it can be done.

Indexed event arguments are stored in topics, and must be of fixed size; Solidity hashes the string and stored it in the topic. Filtering by the hash value of string would work. 

However, hash functions are one-way. We cannot decode the hashes of a specific address to get its list of names. Hence, I decided to not index string in my revised code. 

Thanks for pointing this out, learnt something new about events.

reference: 
https://ethereum.stackexchange.com/questions/6840/indexed-event-with-string-not-getting-logged